### PR TITLE
attempt to use `iproxy` to support iOS real device

### PR DESCRIPTION
**Pull request by truongsinh**
see #39, this change is live on https://www.npmjs.com/package/appium-flutter-driver/v/0.0.24-1

### CODE REVIEW

File: driver/lib/sessions/ios.ts [hunk 1]
```diff
@@ -22,11 +25,22 @@ export const startIOSSession = async (caps) => {
...
 };
 
-export const getObservatoryWsUri = (proxydriver) => {
+export const getObservatoryWsUri = async (proxydriver) => {
   const urlObject = processLogToGetobservatory(proxydriver.logs.syslog.logs);
+  const { udid, realDevice } = proxydriver.opts;
+  if (realDevice) {
+    // @todo check if `brew install usbmuxd` is needed
+    log.info(`Running on iOS real device, doing "iproxy" now`);
+    const cmd = `iproxy ${urlObject.port} ${urlObject.port} ${udid}`;
```
Comment (YueChen-C) [inline driver/lib/sessions/ios.ts:38]: Correct usage of iproxy：
iproxy LOCAL_PORT:DEVICE_PORT -u udid

File: driver/lib/sessions/ios.ts [hunk 2]
```diff
@@ -22,11 +25,22 @@ export const startIOSSession = async (caps) => {
...
+    log.debug(cmd);
+    await execPromise(cmd);
```
Comment (YueChen-C) [inline driver/lib/sessions/ios.ts:40]: the permanent blocking of processes

Comment (shibupanda) [reply driver/lib/sessions/ios.ts:40]: @YueChen-C Do you mean that await and promise needs to be remove. And use below.
child_process.exec(command)

Comment (shibupanda) [reply driver/lib/sessions/ios.ts:40]: I may use spawn here without await and wait for a String "waiting for connection" if it resolve I can send the URL.

Comment (YueChen-C) [reply driver/lib/sessions/ios.ts:40]: good solution

Comment (shibupanda) [reply driver/lib/sessions/ios.ts:40]: Let me apply those and release a bump version

**Comment by ajschwieterman** [comment]
Will this pull request be merged into master and a new tag in the near future?

**Comment by rishav-csenitjsr** [comment]
@shibupanda any update on this? We also need to remove port forwarding for the specific port after the session is completed. I think it is not handled, right?

**Comment by shibupanda** [comment]
>  We also need to remove port forwarding for the specific port after the session is completed. I think it is not handled, right?

Could you please raise a different issue for better tracking.

File: driver/lib/sessions/ios.ts [hunk 3]
```diff
@@ -22,11 +22,25 @@ export const startIOSSession = async (caps) => {
   const observatoryWsUri = getObservatoryWsUri(iosdriver);
   return Promise.all([
     iosdriver,
-    connectSocket(observatoryWsUri, caps.retryBackoffTime, caps.maxRetryCount),
+    connectSocket(await observatoryWsUri, caps.retryBackoffTime, caps.maxRetryCount),
   ]);
 };
 
-export const getObservatoryWsUri = (proxydriver) => {
+export const getObservatoryWsUri = async (proxydriver) => {
```
Comment (hex0cter) [inline driver/lib/sessions/ios.ts:29]: Perhaps await/async is not needed any more? https://github.com/truongsinh/appium-flutter-driver/pull/132

**Pull request merged**